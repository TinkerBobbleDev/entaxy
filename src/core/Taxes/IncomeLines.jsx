import React from 'react'
import PropTypes from 'prop-types'
import { Line } from '@vx/shape'
import { Point } from '@vx/point'
import { Motion, spring } from 'react-motion'
import teal from '@material-ui/core/colors/teal'
import { TaxBrackets, calculateTotalTax } from './lib/TaxBrackets'

const IncomeLines = ({
  income,
  country,
  year,
  region,
  xScale,
  yScale,
  margin,
  height
}) => {
  const yMax = height - margin.top - margin.bottom
  const left = xScale(income)
  const top = yScale(calculateTotalTax(TaxBrackets[country][year], region, income))

  return (
    <Motion
      defaultStyle={{ left, top }}
      style={{
        left: spring(left),
        top: spring(top)
      }}
    >
      {style => (
        <g key="IncomeLines">
          <Line
            key="IncomeLine-vertical"
            from={new Point({ x: style.left, y: yMax })}
            to={new Point({ x: style.left, y: style.top })}
            stroke={teal[500]}
            strokeWidth={1}
          />
          <Line
            key="IncomeLine-horizontal"
            from={new Point({ x: 0, y: style.top })}
            to={new Point({ x: style.left, y: style.top })}
            stroke={teal[500]}
            strokeWidth={1}
          />
          <circle
            cx={style.left - 1}
            cy={style.top + 1}
            r={4}
            fill="white"
            stroke={teal[500]}
            strokeWidth="1.5"
            fillOpacity={1}
            strokeOpacity={1}
          />
        </g>
      )}
    </Motion>
  )
}

IncomeLines.propTypes = {
  income: PropTypes.number.isRequired,
  country: PropTypes.string.isRequired,
  year: PropTypes.number.isRequired,
  region: PropTypes.string,
  xScale: PropTypes.func.isRequired,
  yScale: PropTypes.func.isRequired,
  margin: PropTypes.object.isRequired,
  height: PropTypes.number.isRequired
}

IncomeLines.defaultProps = {
  region: null
}

export default IncomeLines
