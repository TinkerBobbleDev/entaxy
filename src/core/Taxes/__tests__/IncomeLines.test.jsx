import React from 'react'
import renderer from 'react-test-renderer'
import IncomeLines from '../IncomeLines'

describe('IncomeLines', () => {
  it('matches snapshot', () => {
    const margin = {
      top: 20,
      bottom: 40,
      left: 70,
      right: 0
    }
    const xScaleMoch = jest.fn()
    const yScaleMoch = jest.fn()
    const component = renderer.create((
      <IncomeLines
        income={100}
        country="Canada"
        year={2017}
        region="Ontario"
        xScale={xScaleMoch}
        yScale={yScaleMoch}
        margin={margin}
        height={100}
        classes={{ rrsp: {} }}
      />
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
