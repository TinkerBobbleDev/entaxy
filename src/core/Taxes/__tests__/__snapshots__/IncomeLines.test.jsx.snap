// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncomeLines matches snapshot 1`] = `
<g>
  <line
    className="vx-line"
    stroke="#009688"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={40}
    y2={0}
  />
  <line
    className="vx-line"
    stroke="#009688"
    strokeDasharray=""
    strokeWidth={1}
    transform=""
    x1={0}
    x2={0}
    y1={0}
    y2={0}
  />
  <circle
    cx={NaN}
    cy={NaN}
    fill="white"
    fillOpacity={1}
    r={4}
    stroke="#009688"
    strokeOpacity={1}
    strokeWidth="1.5"
  />
</g>
`;
